server/rest/src/main/java/org/infinispan/rest/server/operations/mediatypes/printers/HTMLOutputPrinter.java

```diff
@@ -0,0 +1,28 @@
+package org.infinispan.rest.server.operations.mediatypes.printers;
+
+import java.nio.charset.Charset;
+import java.util.stream.Collectors;
+
+import org.infinispan.CacheSet;
+import org.infinispan.rest.server.operations.exceptions.ServerInternalException;
+import org.infinispan.rest.server.operations.mediatypes.OutputPrinter;
+
+public class HTMLOutputPrinter implements OutputPrinter {
+
+   @Override
+   public byte[] print(String cacheName, CacheSet<?> keys) {
+      return keys.stream()
+            .map(b -> Escaper.escapeHtml(b.toString()))
+            .map(s -> "<a href=\"" + cacheName + "/" + s + "\">" + s + "</a><br/>")
+            .collect(Collectors.joining("<html><body>", "", "</body></html>"))
+            .getBytes(Charset.forName("UTF-8"));
+   }
+
+   @Override
+   public byte[] print(byte[] value) throws ServerInternalException {
+      String html = "<html><body>";
+      html += new String(value);
```

Comment:
Sorry I don't know if I was specific enough for this. What I was saying is that assuming the `byte[]` is already in UTF-8 we could instead statically store the `<html><body>` and `</body></html>` as byte[] encoded as UTF-8. Then when this method is called you just allocate a new byte[] with size equal to all 3 and just use `System.arrayCopy` to copy the contents over.

Something like:
`byte[] result = new byte[head.length + value.length + tail.length];`
`System.arraycopy(head, 0, head.length, result, 0, head.length);`
`System.arraycopy(value, 0, value.length, result, head.length, value.length);`
`System.arraycopy(tail, 0, tail.length, result, head.length + value.length, tail.length);`
`return result`

Where you have

`private static final head byte[] = "<html><body>".getBytes(Charset.forName("UTF-8");`
`private static final tail byte[] = "</body></html>".getBytes(Charset.forName("UTF-8");`

This way you have no encoding required in an invocation since the value byte[] is already in the correct encoding.

Forgive if I messed up my system.array copy arguments :)